src/context/layers/impact.ts

```diff
@@ -168,76 +183,134 @@ export async function fetchImpactLayerData(
-      id: string;
-      baseline: BaselineRecord;
-      feature: GeoJsonBoundary;
-      bounds: ReturnType<typeof bbox>;
-    }[],
-  );
+  // eslint-disable-next-line fp/no-mutation
+  let activeFeatures: GeoJsonBoundary[];
+  if (layer.api) {
+    const wcsUrl = getWCSLayerUrl({
+      layer: hazardLayerDef,
+      extent,
+      date,
+    } as LayerDataParams<WMSLayerProps>);
+    const apiUrl = layer.api.url;
+
+    const apiData = {
+      geotiff_url: wcsUrl,
+      zones_url: layer.api.zonesUrl,
+      group_by: layer.api.groupBy,
+      geojson_out: 'false',
+    };
 
-  // Loop over the features and grab pixels that are contained by the feature.
-  const buckets = matchingFeatures.reduce((acc, { id, feature }) => {
-    const contained = pixelsInFeature(
-      feature,
-      rasters[0],
-      rasters.width,
-      transform,
+    const aggregateData = await fetchApiData(apiUrl, apiData);
+
+    const mergedFeatures = mergeFeaturesByProperty(
+      baselineData.features.features,
+      aggregateData,
+      layer.api.groupBy,
+      operation,
     );
-    return contained.length > 0 ? { ...acc, [id]: contained } : acc;
-  }, {} as { [key: string]: number[] });
 
-  const operation = layer.operation || 'median';
+    // eslint-disable-next-line fp/no-mutation
+    activeFeatures = mergedFeatures.filter(feature => {
+      const scaled = scaleValueIfDefined(
+        feature.properties.stats_median,
+        scale,
+        offset,
+      );
+      return thresholdOrNaN(scaled, layer.threshold);
+    }) as GeoJsonBoundary[];
+  } else {
+    const existingHazardLayer = layerDataSelector(
+      layer.hazardLayer,
+      date,
+    )(getState());
 
-  const activeFeatures = matchingFeatures.reduce(
-    (acc, { id, feature, baseline }) => {
-      const values = buckets[id];
-
-      if (values) {
-        const raw = operations[operation](
-          noData ? values.filter(value => value !== noData) : values,
-        );
-        const scaled = scaleValueIfDefined(raw, scale, offset);
-        const aggregateValue = thresholdOrNaN(scaled, layer.threshold);
-        if (!Number.isNaN(aggregateValue)) {
-          const { properties } = feature;
-
-          const baselineValue = isString(baseline.value)
-            ? parseFloat(baseline.value)
-            : baseline.value;
-          return isNull(baselineValue)
-            ? acc
-            : acc.concat({
-                ...feature,
-                properties: {
-                  ...properties,
-                  [operation]: aggregateValue,
-                  impactValue: baselineValue,
-                },
-              });
-        }
-      }
-      return acc;
-    },
-    [] as GeoJsonBoundary[],
-  );
+    if (!existingHazardLayer) {
+      await dispatch(
+        loadLayerData({
+          layer: hazardLayerDef,
+          extent,
+          date,
+        } as LayerDataParams<WMSLayerProps>),
+      );
+    }
 
+    // eslint-disable-next-line fp/no-mutation
```

Comment:
Why is this `eslint-disable` necessary? It seems like it should be ok with the next line?